index.js

```diff
@@ -0,0 +1,101 @@
+const express = require('express');
+const cors = require('cors');
+const bcrypt = require('bcryptjs');
+const session = require('express-session');
+
+const db = require('./dbHelpers');
+const restrictedRoutes = require('./restrictedRoutes');
+
+const server = express();
+
+// Custom Middleware
+const protect = (req, res, next) => {
+  if (req.session && req.session._id) {
+    next();
+  } else {
+    res.status(400).send('Access denied');
+  }
+};
+
+const restrict = (req, res, next) => {
+  if (req.session && req.session._id) {
+    next();
+  } else {
+    res.status(400).send('YOU SHALL NOT PASS!');
+  }
+};
+
+server.use(express.json());
+server.use(cors());
+server.use(
+  session({
+    name: 'notsession', // default is connect.sid
+    secret: 'nobody tosses a dwarf!',
+    cookie: {
+      maxAge: 1 * 24 * 60 * 60 * 1000
+      // secure: true, // only set cookies over https. Server will not send back a cookie over http.
+    }, // 1 day in milliseconds
+    httpOnly: true, // don't let JS code access cookies. Browser extensions run JS code on your browser!
+    resave: false,
+    saveUninitialized: false
+  })
+);
+
+server.get('/', (req, res) => {
+  res.send('Live server!');
+});
+
+server.use('/api/restricted', restrict, restrictedRoutes);
```

Comment:
Good job building out your server with middleware Jesse 🔥 this line sort of encapsulates how middleware adds more features/abilities to our APIs.